src/Functions/array/array.cpp

```diff
@@ -20,6 +20,7 @@ class FunctionArray : public IFunction
     }
 
     bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForNothing() const override { return false; }
```

Comment:
ok, type for `array(Nothing)` is `Array(Nothing)`

Reply:
Added comment